### PR TITLE
feat(nodejs): make recipe Conan 2.0 ready

### PR DESCRIPTION
**Pull request by stefanpantic-pdftools**
Add version 18.12.0 LTS

Specify library name and version:  **nodejs/18.12.0**

I'm using this recipe in a Conan 2.0 context and wanted to submit my patches to the Conan Center Index as well.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/nodejs/all/conanfile.py [hunk 1]
```diff
@@ -35,17 +36,20 @@ def validate(self):
...
+        copy(self, pattern="*", src=os.path.join(self._source_subfolder, "bin"), dst=os.path.join(self.package_folder, "bin"))
+        copy(self, pattern="node.exe", src=self._source_subfolder, dst=os.path.join(self.package_folder, "bin"))
+        copy(self, pattern="npm", src=self._source_subfolder, dst=os.path.join(self.package_folder, "bin"))
+        copy(self, pattern="npx", src=self._source_subfolder, dst=os.path.join(self.package_folder, "bin"))
 
     def package_info(self):
         self.cpp_info.includedirs = []
         bin_dir = os.path.join(self.package_folder, "bin")
         self.output.info('Appending PATH environment variable: {}'.format(bin_dir))
+        self.runenv_info.prepend_path("PATH", bin_dir)
```
Comment (prince-chrismc) [inline recipes/nodejs/all/conanfile.py:52]: This is done by default and it's not required :)

File: recipes/nodejs/all/test_package/conanfile.py [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+from conan import ConanFile
+from conan.tools.build import cross_building
 
 
 class TestPackageConan(ConanFile):
 
     settings = "os", "arch"
 
+    def requirements(self):
+        self.requires(self.tested_reference_str)
```
Comment (prince-chrismc) [inline recipes/nodejs/all/test_package/conanfile.py:10]:
```suggestion
    def build_requirements(self):
        self.tool_requires(self.tested_reference_str)
```

File: recipes/nodejs/all/test_package/conanfile.py [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
+
     def test(self):
-        if not tools.cross_building(self):
+        if not cross_building(self):
             self.output.info("Node version:")
-            self.run("node --version", run_environment=True)
+            self.run("node --version", env="conanbuild")
```
Comment (prince-chrismc) [inline recipes/nodejs/all/test_package/conanfile.py:15]:
```suggestion
            self.run("node --version")
```

File: recipes/nodejs/all/conanfile.py [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import os
-from conans import ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.tools.files import get, copy
+from conan.errors import ConanInvalidConfiguration
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.53.0"
```
Comment (prince-chrismc) [inline recipes/nodejs/all/conanfile.py:6]: Get and copy were very early so we can drop this :)

```suggestion
required_conan_version = ">=1.47.0"
```

**Review by prince-chrismc — COMMENTED** [review]
This looks like a great start, few suggestions, I think for the "v2" destination this also needs a layout and a package_type as well :)

File: recipes/nodejs/all/conanfile.py (hunk 4 above)
Comment (stefanpantic-smallpdf) [reply recipes/nodejs/all/conanfile.py]: Done

File: recipes/nodejs/all/test_package/conanfile.py (hunk 3 above)
Comment (stefanpantic-smallpdf) [reply recipes/nodejs/all/test_package/conanfile.py]: Removed

File: recipes/nodejs/all/test_package/conanfile.py (hunk 2 above)
Comment (stefanpantic-smallpdf) [reply recipes/nodejs/all/test_package/conanfile.py]: Updated

File: recipes/nodejs/all/conanfile.py (hunk 1 above)
Comment (stefanpantic-smallpdf) [reply recipes/nodejs/all/conanfile.py]: Thanks, removed 😄

**Comment by prince-chrismc** [comment]
It was ported with https://github.com/conan-io/conan-center-index/pull/16966

**Pull request closed without merging**